### PR TITLE
[office-js-preview] Add NAA APIs

### PR DESCRIPTION
**Pull request by AlexJerabek**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://learn.microsoft.com/en-us/samples/officedev/pnp-officeaddins/office-add-in-sso-naa/
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

### CODE REVIEW

**Comment by pkkj** [comment]
The new definition looks good to me. Thanks.

**Comment by AlexJerabek** [comment]
Tagging @davidchesnut for awareness.

File: types/office-js-preview/index.d.ts [hunk 1]
```diff
@@ -3654,6 +3666,35 @@ declare namespace Office {
          */
         forMSGraphAccess?: boolean;
     }
+    /**
+     * Object with user information which can be passed to msal.js.
+     */
+    interface AuthContext {
+        /**
+         * The account unique ID.
```
Comment (Rick-Kirkham) [inline types/office-js-preview/index.d.ts:3674]: I know I'm an old fogy about stacked nouns, but for some reason this one seems especially annoying. 

```suggestion
         * The unique ID of the account.
```

**Review by Rick-Kirkham — APPROVED** [review]
Approving with one suggestion.

File: types/office-js-preview/index.d.ts (hunk 1 above)
Comment (AlexJerabek) [reply types/office-js-preview/index.d.ts:3674]: @Rick-Kirkham Yeah, that does sound weird. Fixed!

**Review by ElizabethSamuel-MSFT — APPROVED** [review]

File: types/office-js-preview/index.d.ts [hunk 2]
```diff
@@ -3654,6 +3666,35 @@ declare namespace Office {
...
+        /**
+         * The full tenant or organizational ID that this account belongs to.
+         */
+        tenantId: string;
+        /**
+         * The user's internet-style login name, based on the Internet standard RFC. Also known as UPN. 
+         */
+        userPrincipalName: string;
+        /**
+         * The identity type by its identity provider (IdP) for this account. "aad" represents an organization account and "msa" represents a live ID account.
```
[the remark this answers is not in the document]
Comment (AlexJerabek) [reply types/office-js-preview/index.d.ts:3686]: That link doesn't resolve. I'll change the text to "Microsoft personal account" - I think that's sufficient. @pkkj - just to clarify, that's the same thing as a Live ID, right?

File: types/office-js-preview/index.d.ts [hunk 3]
```diff
@@ -3594,6 +3594,18 @@ declare namespace Office {
          * @returns Promise to the access token.
          */
         getAccessToken(options?: AuthOptions): Promise<string>;
+        /**
+         * Get the necessary information about the current user. The add-in passes this to MSAL.js to get an access token for the current session.
```
Comment (davidchesnut) [inline types/office-js-preview/index.d.ts:3598]:
```suggestion
         * Gets information about the signed-in user. The add-in can pass this information to the Microsoft authentication library (MSAL.js) to get an access token for the current session.
```

**Review by davidchesnut — APPROVED** [review]
had one comment but looks great! Thanks for doing this!

File: types/office-js-preview/index.d.ts (hunk 3 above)
Comment (AlexJerabek) [reply types/office-js-preview/index.d.ts:3598]: I'll add this to a new PR.

**Pull request merged**